Plugins/Public/MiscellaneousCommands/Main.cpp

```diff
@@ -0,0 +1,256 @@
+	{
+		if (set_scCfgFile.length()>0)
+			LoadSettings();
+	}
+	else if (fdwReason == DLL_PROCESS_DETACH)
+	{
+	}
+	return true;
+}
+
+/// Hook will call this function after calling a plugin function to see if we the
+/// processing to continue
+EXPORT PLUGIN_RETURNCODE Get_PluginReturnCode()
+{
+	return returncode;
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Structures and Declarations
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// See Main.h for any struct/class defs.
+// This is just for declarations
+
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Loading Settings
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void LoadSettings()
+{
+	returncode = DEFAULT_RETURNCODE;
+	// Reserved for future use
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Functions
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+bool CheckIsInBase(uint iClientID)
+{
+	uint iBaseID;
+	pub::Player::GetBase(iClientID, iBaseID);
+	if (!iBaseID)
+	{
+		PrintUserCmdText(iClientID, L"You must be in a base to use this command.");
+		return false;
+	}
+
+	return true;
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Command Functions
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// /refresh - Updates the timestamps of the character file for all the ships on the account.
+bool UserCmd_RefreshAccounts(uint iClientID, const wstring &wscCmd, const wstring &wscParam, const wchar_t *usage)
+{
+	if (!CheckIsInBase(iClientID))
+		return true;
+
+	FILETIME ft;
+	SYSTEMTIME st;
+	GetLocalTime(&st);
+	SystemTimeToFileTime(&st, &ft);
+	char toWrite[128];
+	sprintf_s(toWrite, "%u,%u", ft.dwHighDateTime, ft.dwLowDateTime);
+	
+	CAccount *acc = HkGetAccountByClientID(iClientID);
+	CAccountListNode *accountList = acc->pFirstListNode;
+	PrintUserCmdText(iClientID, L"Refreshing Accounts:");
+	PrintUserCmdText(iClientID, L""); // Line break to make it look nice
+	int iAccountsRefreshed = 0; // Number to keep track of
+	HK_ERROR err; // If it errors we want to inform them
+
+	// Save the current char in case for whatever reason they haven't been saved already
+	// Prevents any potential data loss.
+	HkSaveChar(reinterpret_cast<const wchar_t*>(Players.GetActiveCharacterName(iClientID)));
+
+	// We need to keep track of the first character name lest it loops forever.
+	wstring wscFirstCharacterName;
+	while (accountList)
+	{
+		// We need to make sure the character name is not null (which is is sometimes because FLHook?)
```

Comment:
Bare in mind this is not FLHook code we're dealing with here - CAccount is FL code itself that we import.